sympy/physics/units/dimensions.py

```diff
@@ -572,3 +572,74 @@ def is_consistent(self):
         # dimensions
         # in vector language: the set of vectors do not form a basis
         return self.inv_can_transf_matrix.is_square
+    def getDimensionlessNumbers(self, list_of_derived_Quantities 
+        # input list of derived quantities in the
+        # form of a python dictionary of tuples (key, item) where 'key' is
+        # the name of the physical quantity and 'item' is an instance of
+        # Dimension class which possesses the dimensions of the quantity.
+        ):
+        """ This function is designed to take in a set of physical variables (all 
+    	essentially dimensionful) and returns one set of all possible dimensionless
+	    quantities, as guided by the Buckingham's pi theorem (a generalization of 
+	    Rayleigh's method of dimensional analysis .....)
+	    
+        Please refer and run to the unit testing file '/units/test/test_dimensionLess.py' for
+            the format of the input .... As a heads up, we first need to define
+            a global set of base dimensions and with those we can define the 
+            derived quantities .....
+        """
+        ExponentMatrix = []
+        IndexDict = { # Dimension names borrowed from print(dimsys_SI)
+            'length' : 0,
+            'time' : 1,
+            'mass' : 2,
+            'current': 3,
+            'temperature': 4,
+            'amount_of_substance': 5,
+            'luminous_intensity': 6
+            }
+        """
+            We extract the exponent matrix from the dimensional dependencies of the
+            quantities
+        """
+        for quantity in list_of_derived_Quantities:
+            ElementForQuantity = [0 for k in range(7)]
+            dimensions = dimsys_SI.get_dimensional_dependencies(
```

Comment:
You are using a reference to `dimsys_SI`, which is an instance of `Dimensions`, inside the class `Dimension`. It makes the algorithm dependent on the dimension system of SI units only. The units module should be independent of the chosen unit/dimension system.

What about `self.get_dimensional_dependencies` instead?

Reply:
@Upabjojr thanks much Francesco. So may I include it as an edit here or would it be a separate commit somewhere?

Reply:
What do you mean by _edit_? An edit will create a new commit.